### ISSUE TITLE
[no-extra-parens] false negative for type cast

### ISSUE THREAD

**Issue by awvalenti** [typescript-eslint/typescript-eslint#1872]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-extra-parens": ["error"]
  }
}
```

```TS
let a: number | string = 'text'
if (Math.random() < 0.5) a = 1
// eslint-disable-next-line no-console
console.log((((((a as string))))).toLowerCase())
```

**Expected Result**
Detection of extra parens

**Actual Result**
No error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^2.21.0` |
| `@typescript-eslint/parser`        | `^2.21.0` |
| `TypeScript`                       | `^3.7.2` |
| `ESLint`                           | `^6.8.0` |
| `node`                             | `v13.1.0` |
| `npm`                              | `6.12.1` |

**Comment by bradzacher** [comment]
The rule extension is a terrible hack, so it's not surprising there's some edge cases.
Happy to accept a PR, if it's important to you.

----

Formatting rules are hard to maintain and hard to support.
I ***strongly*** recommend using an auto-formatting tool like `prettier` instead.

**Comment by bradzacher** [comment]
Looking a little into this - you're misunderstanding what the rule does.

To quote the rule docs:
> This rule restricts the use of parentheses to only where they are necessary.

This rule does not error when there are _too many_ parens, it just ensures you don't have them when they're not needed.

Which is why the following code passes in the base rule:
```js
(((((x in y))))).foo
```
https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwibm8tZXh0cmEtcGFyZW5zXCI6IFwiZXJyb3JcIiAqL1xuKCgoKCh4IGluIHkpKSkpKS5mb28iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjksInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnsianN4Ijp0cnVlfX0sInJ1bGVzIjp7fSwiZW52Ijp7fX19

**Comment by awvalenti** [comment]
Thanks for your attention!
This behavior makes no sense to me, but is in fact according to the spec.
